client/src/components/ElementActions/SaveAction.js

```diff
@@ -1,128 +1,56 @@
+import React, { useContext, useEffect } from 'react';
 import AbstractAction from 'components/ElementActions/AbstractAction';
-import backend from 'lib/Backend';
 import i18n from 'i18n';
-import { loadElementSchemaValue } from 'state/editor/loadElementSchemaValue';
-import { loadElementFormStateName } from 'state/editor/loadElementFormStateName';
-import { initialize } from 'redux-form';
+import { ElementContext } from 'components/ElementEditor/Element';
 
-/**
- * Using a REST backend, serialize the current form data and post it to the backend endpoint to save
- * the inline edit form's data for the current block.
- */
 const SaveAction = (MenuComponent) => (props) => {
-  if (!props.expandable || props.type.broken) {
-    // Some elemental blocks can not be edited inline (e.g. User form blocks)
-    // We don't want to add a "Save" action for those blocks.
-    return (
-      <MenuComponent {...props} />
-    );
-  }
+  const {
+    doSaveElement,
+    formHasRendered,
+    onSaveButtonClick,
+    onAfterSave,
+    submitForm
+  } = useContext(ElementContext);
 
   const handleClick = (event) => {
     event.stopPropagation();
+    onSaveButtonClick();
+  };
 
-    const { element, type, securityId, formData, reinitialiseForm } = props;
-    const { jQuery: $ } = window;
-    const noTitle = i18n.inject(
-      i18n._t(
-        'ElementHeader.NOTITLE',
-        'Untitled {type} block'
-      ),
-      { type: type.title }
-    );
-
-    const endpointSpec = {
-      url: loadElementSchemaValue('saveUrl', element.id),
-      method: loadElementSchemaValue('saveMethod'),
-      payloadFormat: loadElementSchemaValue('payloadFormat'),
-      defaultData: {
-        SecurityID: securityId
-      },
-    };
-
-    const endpoint = backend.createEndpointFetcher(endpointSpec);
-    endpoint(formData)
-      .then(() => {
-        // Update the Apollo query cache with the new form data
-        const { apolloClient } = window.ss;
-
-        apolloClient.queryManager.reFetchObservableQueries();
-        reinitialiseForm(formData);
-
-        const preview = $('.cms-preview');
-        preview.entwine('ss.preview')._loadUrl(preview.find('iframe').attr('src'));
-
-        const newTitle = formData ? formData[`PageElements_${element.id}_Title`] : null;
-        $.noticeAdd({
-          text: i18n.inject(
-            i18n._t(
-              'ElementSaveAction.SUCCESS_NOTIFICATION',
-              'Saved \'{title}\' successfully'
-            ),
-            { title: newTitle || noTitle }
-          ),
-          stay: false,
-          type: 'success'
-        });
-      })
-      .catch(() => {
-        $.noticeAdd({
-          text: i18n.inject(
-            i18n._t(
-              'ElementSaveAction.ERROR_NOTIFICATION',
-              'Error saving \'{title}\''
-            ),
-            { title: element.Title || noTitle }
-          ),
-          stay: false,
-          type: 'error'
-        });
-      });
+  const saveElement = () => {
```

Comment:
This is kind of unnecessary convoluted in that formHasRendered, doSaveElement, submitForm and onAfterSave() all come in via Context set on Element.js.  We could actually do just about everything within Element.js.

The reason it's like this is to mirror PublishAction.js which does have some GraphQL HOC logic that's not available in Element.js